### PR TITLE
DC-454: Create flows configuration to be send to minion

### PR DESCRIPTION
**Pull request by freddy61025**
## Jira link(s)
- https://opennms.atlassian.net/browse/DC-454

## Flagged for review

The minion receiver will be done in https://opennms.atlassian.net/browse/DC-455

## Checklist
* [x] Follows Horizon Stream's [development guidelines.](https://github.com/OpenNMS/horizon-stream/wiki/Development-Guidelines)
* [x] Appropriate reviewer(s) have been selected.
* [x] Jira issue(s) have been updated to "In Review".
* [x] Includes [appropriate tests.](https://github.com/OpenNMS/horizon-stream/wiki/Test-Strategy)
* [x] Documentation has been updated as necessary.

### CODE REVIEW

**Review by FrancescoBuc86 — COMMENTED** [review]
It seems fine, just try to get ride of the 4 code smells..

File: inventory/src/main/java/org/opennms/horizon/inventory/service/FlowsConfigService.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+@Service
+@RequiredArgsConstructor
+public class FlowsConfigService {
+    private static final Logger LOG = LoggerFactory.getLogger(FlowsConfigService.class);
+    private final MonitoringLocationService monitoringLocationService;
+    private final TaskSetManager taskSetManager;
+    private final TaskSetPublisher taskSetPublisher;
+
+    @EventListener(ApplicationReadyEvent.class)
+    public void sendTrapConfigToMinionAfterStartup() {
```
Comment (cgorantla) [inline inventory/src/main/java/org/opennms/horizon/inventory/service/FlowsConfigService.java:62]: `sendFlowConfigToMinionAfterStartup()`

File: inventory/src/main/java/org/opennms/horizon/inventory/service/MonitoringSystemService.java [hunk 2]
```diff
@@ -53,7 +54,7 @@ public Optional<Long> addMonitoringSystemFromHeartbeat(HeartbeatMessage message,
                 location.setTenantId(tenantId);
                 var newLocation = locationRepository.save(location);
                 trapConfigService.sendTrapConfigToMinion(newLocation.getTenantId(), newLocation.getLocation());
-
+                flowsConfigService.sendFlowsConfigToMinion(newLocation.getTenantId(), newLocation.getLocation());
```
Comment (cgorantla) [inline inventory/src/main/java/org/opennms/horizon/inventory/service/MonitoringSystemService.java:57]: ConfigUpdateService from  PR #567 might be useful here.

File: shared-lib/protobuf/src/main/proto/flows-config.proto [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+message Filter {
+  string content = 1;
+}
+
+message RrdDefinition {
+}
+
+// RRD related settings for a given telemetry protocol package.
+message Rrd {
```
Comment (cgorantla) [inline shared-lib/protobuf/src/main/proto/flows-config.proto:90]: we can remove RRD definitions, we will not be using RRD in Stream

File: inventory/src/main/java/org/opennms/horizon/inventory/service/FlowsConfigService.java (hunk 1 above)
Comment (freddy61025) [reply inventory/src/main/java/org/opennms/horizon/inventory/service/FlowsConfigService.java]: Fixed

File: inventory/src/main/java/org/opennms/horizon/inventory/service/MonitoringSystemService.java (hunk 2 above)
Comment (freddy61025) [reply inventory/src/main/java/org/opennms/horizon/inventory/service/MonitoringSystemService.java]: I rebased with develop.

File: shared-lib/protobuf/src/main/proto/flows-config.proto (hunk 3 above)
Comment (freddy61025) [reply shared-lib/protobuf/src/main/proto/flows-config.proto]: Removed

**Review by FrancescoBuc86 — APPROVED** [review]

File: inventory/src/main/java/org/opennms/horizon/inventory/service/FlowsConfigService.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    }
+
+    public void sendFlowsConfigToMinion(String tenantId, String location) {
+        FlowsConfig flowsConfig = readFlowsConfig();
+        publishFlowsConfig(tenantId, location, flowsConfig);
+    }
+
+    private void publishFlowsConfig(String tenantId, String location, FlowsConfig flowsConfig) {
+        TaskDefinition taskDefinition = TaskDefinition.newBuilder()
+            .setId("flows-config")
```
Comment (patrick-schweizer) [inline inventory/src/main/java/org/opennms/horizon/inventory/service/FlowsConfigService.java:75]: just a question: how do we synchronize the naming to the receiver? Not sure if we can / should externalize the ids

File: inventory/src/main/java/org/opennms/horizon/inventory/service/FlowsConfigService.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+public class FlowsConfigService {
+    private static final Logger LOG = LoggerFactory.getLogger(FlowsConfigService.class);
+    private final MonitoringLocationService monitoringLocationService;
+    private final TaskSetPublisher taskSetPublisher;
+
+    @EventListener(ApplicationReadyEvent.class)
+    public void sendFlowConfigToMinionAfterStartup() {
+        List<MonitoringLocationDTO> allLocations = monitoringLocationService.findAll();
+
+        for (MonitoringLocationDTO dto : allLocations) {
```
Comment (patrick-schweizer) [inline inventory/src/main/java/org/opennms/horizon/inventory/service/FlowsConfigService.java:63]: not sure about the desired behavior: if one call to sendFlowsConfigToMinion() fails, the rest won't be executed anymore.

File: inventory/src/test/resources/flows-config.json [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    "name": "Netflow-5-UDP-8877",
+    "className": "org.opennms.netmgt.telemetry.listeners.UdpListener",
+    "enabled": true,
+    "parameters": [{
+      "key": "port",
+      "value": "8877"
+    }],
+    "parsers": [{
+      "name": "Netflow-5-Parser",
+      "className": "org.opennms.netmgt.telemetry.protocols.netflow.parser.Netflow5UdpParser",
```
Comment (patrick-schweizer) [inline inventory/src/test/resources/flows-config.json:12]: Do we need to adopt the package structure according to the changes in _DC-496 Clean up package structure_ ?

File: shared-lib/protobuf/src/main/proto/flows-config.proto [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+syntax = "proto3";
+
+package opennms.sink.flows;
+option java_multiple_files = true;
+option java_package = "org.opennms.sink.flows.contract";
+
+// rename from TelemetrydConfig
```
Comment (patrick-schweizer) [inline shared-lib/protobuf/src/main/proto/flows-config.proto:34]: remove?

**Review by patrick-schweizer — COMMENTED** [review]
A few comments, not sure how relevant...

File: inventory/src/main/java/org/opennms/horizon/inventory/service/FlowsConfigService.java (hunk 4 above)
Comment (freddy61025) [reply inventory/src/main/java/org/opennms/horizon/inventory/service/FlowsConfigService.java:75]: It seems the current way is hardcode in both side. I would like to do class name to kebab case at both side. I will think about it when i am working DC-455.

File: shared-lib/protobuf/src/main/proto/flows-config.proto (hunk 7 above)
Comment (freddy61025) [reply shared-lib/protobuf/src/main/proto/flows-config.proto]: removed

File: inventory/src/test/resources/flows-config.json (hunk 6 above)
Comment (freddy61025) [reply inventory/src/test/resources/flows-config.json]: sure although it won't hurt anything but align make it cleaner.

File: inventory/src/main/java/org/opennms/horizon/inventory/service/FlowsConfigService.java (hunk 5 above)
Comment (freddy61025) [reply inventory/src/main/java/org/opennms/horizon/inventory/service/FlowsConfigService.java:63]: exception captured.

**Review by patrick-schweizer — APPROVED** [review]
looks now good to me!

**Review by cgorantla — APPROVED** [review]
LGTM!

**Pull request merged**